lib/data/data_source/remote/database_remote_source_impl.dart

```diff
@@ -96,4 +99,30 @@ class DatabaseRemoteSourceImpl implements DatabaseRemoteSource {
       throw ApiException(Errors.somethingWentWrong);
     }
   }
+  @override
+  Future<Success> updateUserProfile(UserProfileDto dto) async {
+    try {
+      final userId = firebaseAuth.currentUser?.uid;
+      if (userId != null) {
+        final doc = await firestore.collection("users").doc(userId).get();
+        if (doc.exists) {
+          final UserProfileDto existingUserDoc = UserProfileDto.fromJson(doc.data()!);
+          final updatedUserDoc = existingUserDoc.copyWith(
+            name: dto.name.isNotEmpty ? dto.name : existingUserDoc.name,
+            userId: dto.userId.isNotEmpty ? dto.userId : existingUserDoc.userId,
+            initialLanguage: dto.initialLanguage.isNotEmpty ? dto.initialLanguage : existingUserDoc.initialLanguage,
+            userFolders: dto.userFolders.isNotEmpty ? dto.userFolders : existingUserDoc.userFolders,
+          );
+          await firestore.collection("users").doc(userId).set(updatedUserDoc.toJson());
```

Comment:
You can use the update method to update an existing doc:

`database.collection("your_collection").doc("your_doc_id").update({
  "fieldToUpdate": newValue,
  // Add other fields to update as needed
});`